src/mimeparser.rs

```diff
@@ -3820,4 +3820,25 @@ Content-Disposition: reaction\n\
 
         Ok(())
     }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_long_and_duplicated_filenames() -> Result<()> {
+        let mut tcm = TestContextManager::new();
+        let alice = tcm.alice().await;
+        let bob = tcm.bob().await;
+
+        bob.set_config(Config::Displayname, Some("bob")).await?;
+
+        let msg = tcm.send_recv_accept(&alice, &bob, "hi").await;
+        let msg = &bob.send_text(msg.chat_id, "hi").await;
```

Comment:
Maybe check it's encrypted? And also that there's no displayname in the unencrypted part

Reply:
second thing is checked below, added a check for encryption